### ISSUE TITLE
"Too much recursion" on nearly-empty lists

### ISSUE THREAD

**Issue by jfengel** [ericgio/react-bootstrap-typeahead#142]
The following code:

`<Typeahead options={[""]}>`

It renders fine, but when I click on it, it delays a few moments before declaring "Too much recursion". Other uses of Typeahead with sensible options in the same program works fine.

**Comment by ericgio** [comment]
Thanks for the report, I'll look into this.

**Comment by ericgio** [comment]
I don't have a definitive solution yet, but here's a what I've discovered so far:

This only happens when the **first** option is an empty string:
```jsx
// These cases cause the error:
['']
['', 'foo']

// These cases do not:
['foo', '']
[{label: ''}]
```
It appears that having an empty string as the first option causes React to continuously re-render the component, thus causing the recursion/call stack error.

So, this seems like a pretty specific case that really shouldn't happen unless you're intentionally passing an empty string as your first data option.

**Comment by jfengel** [comment]
Yeah, it's a weird corner-case. I probably wouldn't have seen it except for some bad user data (which I have since filtered out). But I thought it was a very odd way for the code to respond, and it might be worth looking into, Just In Case.

(Having very much success with the component, BTW. Thanks!)

**Comment by Tauka** [comment]
I report facing this issue as well. My use case is trying to empty the input using `clear()` inside of onChange prop function. I use onChange as onClick for items to navigate to another router url, so I tried to empty it right after router transition occurs.

Here is my function for onChange prop:
```javascript
goToQuestionForm(selected) {

		//if there is a input, navigate to its url
		if (selected[0] != undefined) {
			this.props.router.push(`question/${selected[0].id}`);
		}

		//clear the input
		this.typeahead.getInstance().clear();
	}
```

Hope it helps. Good luck!

**Comment by ericgio** [comment]
@Tauka: Yours is a different issue. `clear` calls `onChange` so you're creating an infinite loop.

**Comment by ericgio** [comment]
Fixed in v2.0.0-alpha.1
